crowdsec-docs/unversioned/bouncers/haproxy_spoa.mdx

```diff
@@ -0,0 +1,400 @@
+  ]
+}>
+<TabItem value="haproxy_debian">
+
+```bash
+sudo apt install crowdsec-haproxy-spoa-bouncer
+```
+
+</TabItem>
+
+<TabItem value="haproxy_rhel">
+
+```bash
+sudo dnf install crowdsec-haproxy-spoa-bouncer
+```
+
+</TabItem>
+
+</Tabs>
+
+
+## Configuration
+
+If you are using packages, and have a lapi on the same server the following
+configuration file `/etc/crowdsec/bouncer/crowdsec-spoa-bouncer.yaml` should
+already be in a working state.
+
+You can always edit the configuration file at `/etc/crowdsec/bouncer/crowdsec-spoa-bouncer.yaml`:
+
+```yaml
+log_mode: file
+log_dir: /var/log/
+log_level: info
+log_compression: true
+log_max_size: 100
+log_max_backups: 3
+log_max_age: 30
+
+update_frequency: 10s
+api_url: http://127.0.0.1:8080/
+api_key: ${API_KEY}
+insecure_skip_verify: false
+
+workers:
+  - name: spoa1
+    listen_addr: 0.0.0.0:9000
+    listen_socket: /run/crowdsec-spoa/spoa-1.sock
+
+worker_user: crowdsec-spoa
+worker_group: crowdsec-spoa
+
+asn_database_path: /var/lib/crowdsec/data/GeoLite2-ASN.mmdb
+city_database_path: /var/lib/crowdsec/data/GeoLite2-City.mmdb
+
+admin_socket: /run/crowdsec-spoa-admin.sock
+
+prometheus:
+  enabled: true
+  listen_addr: 127.0.0.1
+  listen_port: 60601
+```
+
+You can get a workable configuration by using the yaml above and getting and api key by:
+```bash
+sudo cscli bouncers add mybouncer
+API key for 'bouncertest':
+
+   JdVa7DKBM35gPDAR014pH/55l38fxLGt02NPPnZgLQI
+
+Please keep this key since you will not be able to retrieve it!
+```
+
+## HAProxy Configuration
+
+If you are using packages, you will find the haproxy configuration snippets in
+`/usr/share/doc/crowdsec-haproxy-spoa-bouncer/examples`.
+
+### SPOE Filter
+
+Add a SPOE agent configuration to /etc/haproxy/crowdsec.cfg:
+
+```
+[crowdsec]
+spoe-agent crowdsec-agent
+    messages    crowdsec-ip crowdsec-http
+
+    option      var-prefix      crowdsec
+    option      set-on-error    error
+    timeout     hello           100ms
+    timeout     idle            30s
+    timeout     processing      500ms
+    use-backend crowdsec-spoa
+    log         global
+
+## This message is used to customise the remediation from crowdsec-ip based on the host header
+spoe-message crowdsec-http
+    args remediation=var(txn.crowdsec.remediation) crowdsec_captcha_cookie=req.cook(crowdsec_captcha_cookie) id=unique-id host=hdr(Host) method=method path=path query=query version=req.ver headers=req.hdrs body=req.body url=url ssl=ssl_fc
+    event on-frontend-http-request
+
+## This message should be the first to trigger in the chain
+spoe-message crowdsec-ip
+    args id=unique-id src-ip=src src-port=src_port
+    event on-client-session
+
+```
+
+If you installed the haproxy spoe bouncer through package, you will find this
+configuration file in `/usr/share/docs/crowdsec-haproxy-spoa-bouncer/examples`
+
+This crowdsec spoe agent configuration is then referenced in the main haproxy configuration file.
```

Comment:
specify the path of "the main haproxy configuration file". Should it be added at the end ? at the beginning ? or it doesn't matter ?